packages/core/src/interpreter.ts

```diff
@@ -464,6 +463,14 @@ export class Interpreter<
       ? this.logic.getSnapshot(this._state)
       : this._state;
   }
+
+  public reset(
+    persistedState: PersistedStateFrom<TLogic> | undefined = this.options.state
```

Comment:
> So should we persist input after all? Just for **potential** recovery? I think this is somewhat wrong (or at least non-ideal) because that raises the used memory unconditionally, for every machine in the application and ofc increases the DB size for persisted data.

I think so, at least if the "resume" supervision option is chosen.

If memory size is a concern, the developer can choose a different option, which won't persist `input`.

Reply:
> I think so, at least if the "resume" supervision option is chosen.

`input` is not relevant for "resume" but rather for "start from scratch".

Reply:
Edited

Reply:
> If memory size is a concern, the developer can choose a different option, which won't persist input.

If the user *chooses* the supervision strategy, I think it's a fair game to persist `input` for the purpose of recovery. 

However, the proposed `reset` method is always available so the developer doesn't choose anything by default. With `reset` there is an implied strategy and I think that we can both agree that for such a method the "start from scratch" strategy is the most sensible default. 

That breaks the principle of least surprise when it comes to the memory management. Imagine a file uploader machine that spawns many machines where each one is responsible for uploading a single file. Does it make sense to pass `File`/`Blob` as `input` to the spawned child? Definitely. A file upload machine likely won't keep an explicit reference to that `File`/`Blob` - especially once the upload is finished. Why would it? But yet we would hold to it just because it was passed part of `input`. Does the user expected that? I wouldn't.

Reply:
Then "start from scratch" shouldn't be the default, if memory is a concern.

Either we make it clear in the docs that, if it's the default, `input` will be persisted, or we choose a different default.

What do you think should be the solution here?

Reply:
Potential solution:

```ts
// same interpreter options
interpret(machine, { input: someInput });

const previousInput = await db.read(...);

actor.reset({
  input: previousInput // developer's responsibility
});

actor.reset({
  state: persistedState // deep recursive persisted state
});
```